### PR TITLE
ACS-6869: Fixing filter logic.

### PR DESCRIPTION
**Pull request by mpichura**
This PR fixes aspect filtering logic for a case when not all aspects present on a node are present in allow list (but some are).
That seemed to be a problem with integration test (as the unit test tested the wrong logic but integration test was testing the correct one).
Edit:
I reverted filtering logic. Only changes now are test fixes.

### CODE REVIEW

**Review by krdabrowski — APPROVED** [review]
Nice fix!

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplier.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public boolean applyFilter(RepoEvent<DataAttributes<NodeResource>> repoEvent, Fi
 
     private boolean isAllowed(Set<String> aspectNames, List<String> allowed)
     {
-        return CollectionUtils.isEmpty(allowed) || allowed.stream().anyMatch(aspectNames::contains);
+        return allowed.isEmpty() || !aspectNames.isEmpty() && new HashSet<>(allowed).containsAll(aspectNames);
```
Comment (tpage-alfresco) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplier.java:62]: I don't think this is correct:
```
aspectNames: a, b
allowed: a, c
allowed.containsAll(aspectNames) -> false
```
In this case I'd expect the presences of aspect `a` to qualify the node.

File: live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplierTest.java [hunk 2]
```diff
@@ -160,7 +160,7 @@ void shouldNotFilterOutWhenAspectInAllowedAndEmptyDenied()
         boolean result = objectUnderTest.applyFilter(mockRepoEvent, mockFilter);
 
         // then
-        assertTrue(result);
+        assertFalse(result);
```
Comment (tpage-alfresco) [inline live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplierTest.java:163]: I think this should be `assertTrue`.

**Review by tpage-alfresco — CHANGES_REQUESTED** [review]
I think the original logic was correct. Perhaps the integration test needs updating?

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplier.java (hunk 1 above)
Comment (mpichura) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplier.java]: Reverted

File: live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplierTest.java (hunk 2 above)
Comment (mpichura) [reply live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplierTest.java]: Reverted

**Review by tpage-alfresco — APPROVED** [review]
LGTM!

**Pull request merged**